src/pyodide/internal/snapshot.js

```diff
@@ -412,3 +417,15 @@ export function finishSnapshotSetup(pyodide) {
     });
   }
 }
+
+/**
+ * Store the memory snapshot in the ArtifactBundler so that the validator can read it out.
+ * Needs to happen at the top level because the validator does not perform requests.
+ */
+export function maybeStoreMemorySnapshot() {
+  if (ArtifactBundler.isEwValidating()) {
+    ArtifactBundler.storeMemorySnapshot(getMemoryToUpload());
+  } else if (SHOULD_SNAPSHOT_TO_DISK) {
+    DiskCache.put("snapshot.bin", getMemoryToUpload());
```

Comment:
Not blocking, but I think we should name the snapshot something different for each worker, since eventually we want to load the snapshot from disk cache in addition to storing it. I think it may make sense to turn the `python-save-snapshot` flag into a string where the string is the file name, or use some other unique identifier for the worker.

Reply:
Yeah for now I just want some way to get snapshots onto disk that I can use for tests. Note that this doesn't even implement a way to load the snapshot in workerd, only to save it. Which is probably much higher priority than giving the files distinct names...